### ISSUE TITLE
filename in release

### ISSUE THREAD

**Issue by TJM** [jfrog/artifactory-secrets-plugin#57]
... as per the changelog in v0.2.12:

```md
IMPROVEMENTS:

- Plugin now reports its version to Vault server. You can see it with `vault plugin list` command.
- Remove version number from the binary file name (now `artifactory-secrets-plugin`, vs `artifactory-secrets-plugin_v0.2.6`) now that it registers as 'versioned' plugin with Vault server.
- Update README on how to register plugin to reflect this change of binary name.
- Update Makefile to use GoRelease (same as GitHub Action) to build binary for development process.
```

The problem is that when upgrading the plugin on a cluster, you will need both binaries on the system at the same time, so you will need to rename it to have the version in the name or do something funky like _new or _old ??

I can handle this in my packer script, but it was easier to just have the name be the version.

Or.. maybe I am doing something wrong? :)

**Comment by alexhung** [comment]
The way I understand (and assume) is that when Vault server mount/enable a plugin, it starts a new process(?) thus keeping it in memory. I can be wrong though. Overwriting the binary with new version won't affect existing mounted plugin. And to distinguish between old and new during registration, the option `-version` would be used.

(Of course one should make a backup copy (if not already) of the existing binary before copying in the new version.)

I'm open to revert back to including version in file name if it makes life easier. The reason for the switch is to follow common convention where binary name *does not* include version.

**Comment by TJM** [comment]
I am not saying to change for me. Perhaps my vault cluster in a Google MIG is overly complex. I actually think the upgrade instructions say to backup the old binary, but maybe that was to be able to move back to the previous version. I think it said to not overwrite the binary, but it seems like it could stop/start that process at any time, so I had planned to keep the full name.

**Comment by alexhung** [comment]
True. When I examine a few plugins in the both official and partner plugins, all of the binaries do not contain the version string.

**Comment by TJM** [comment]
I guess I will use my script above to keep the version string... I wonder if something like that would be useful? It is part of our packer scripts to build a vault image.

**Comment by alexhung** [comment]
Feel free to open a PR to add your script to the project (in `./scripts` directory?).

**Comment by TJM** [comment]
Interesting idea... it could for sure be in scripts or maybe "examples" or something? I could actually include more stuff (Terraform)

**Comment by alexhung** [comment]
'examples' is better fit, along with references to them in README
